src/webui/api/rsscontroller.cpp

```diff
@@ -91,6 +94,30 @@ void RSSController::itemsAction()
     setResult(jsonVal.toObject());
 }
 
+void RSSController::markItemAsReadAction()
+{
+    requireParams({"itemPath"});
+
+    const QString itemPath {params()["itemPath"]};
+    RSS::Item *item = RSS::Session::instance()->itemByPath(itemPath);
+    if (item)
+        item->markAsRead();
+}
+
+void RSSController::markArticleAsReadAction()
+{
+    requireParams({"itemPath", "articleId"});
+
+    const QString itemPath {params()["itemPath"]};
+    const QString articleId {params()["articleId"]};
+
+    RSS::Feed *item = (RSS::Feed*) RSS::Session::instance()->itemByPath(itemPath);
```

Comment:
We forbid C-style cast in favor of C++-style one.
Anyway it is incorrect since Folder can be here instead. You should use qobject_cast.